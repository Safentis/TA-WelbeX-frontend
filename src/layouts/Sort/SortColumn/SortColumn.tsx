import React, { FC } from 'react';

import './SortColumn.css';
import { Props } from './SortColumn.interface';
import { useAppContext } from '../../../application/App/App.context';

/**
 * SortColumn component
 * @returns {React.ReactElement} - returns react element
 */

const SortColumn: FC<Props> = ({}): React.ReactElement => {
  //* ----------------------------------------------------
  //* In this case we used context
  const { state, setState } = useAppContext();
  
  const handleChange = (event: React.ChangeEvent): void => {
    //* We are creating copy of state;
    const newState = { ...state };

    //* After we are choosing of the column value
    const targer = event.target as HTMLInputElement;
    const column = newState.filter.column;
    const name: string = targer.value;

    //* If it's value we are seting true
    //* else false
    for (let col in column) {
      (col !== name)
        ? column[col] = false
        : column[name] = true;
    }

    //* And finaly, we are seting new state
    setState(newState);
  };

  return (
    <ul className="sort-column sort__nested">
      <li className="sort-column__item">
        <input
          type="radio"
          value="name"
          name="col"
          checked={state.filter.column['name']}
          onChange={handleChange}
        />
      </li>
      <li className="sort-column__item">
        <input
          type="radio"
          value="number"
          name="col"
          checked={state.filter.column['number']}
          onChange={handleChange}
        />
      </li>
      <li className="sort-column__item">
        <input
          type="radio"
          value="distance"
          name="col"
          checked={state.filter.column['distance']}
          onChange={handleChange}
        />
      </li>
    </ul>
  );
};

export default SortColumn;
