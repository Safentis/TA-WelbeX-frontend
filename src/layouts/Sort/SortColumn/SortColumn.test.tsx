import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import SortColumn from './SortColumn';

import { Props } from './SortColumn.interface';

const setUp = (props: Props) => {
  return shallow(<SortColumn {...props} />);
};

describe('<SortColumn />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;

  beforeEach(() => {
    component = setUp({});
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 ul element', () => {
      expect(component.find('ul')).toBeTruthy();
    });

    it('has 3 li element', () => {
      expect(component.find('li').length === 3).toBeTruthy();
    });

    it('has 1 input element', () => {
      expect(component.find('input').length === 3).toBeTruthy();
    });
  });
});
