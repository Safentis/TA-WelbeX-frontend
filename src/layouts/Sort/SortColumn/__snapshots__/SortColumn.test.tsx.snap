// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortColumn /> component does match to the snapshot 1`] = `
<ul
  className="sort-column sort__nested"
>
  <li
    className="sort-column__item"
  >
    <input
      checked={true}
      name="col"
      onChange={[Function]}
      type="radio"
      value="name"
    />
  </li>
  <li
    className="sort-column__item"
  >
    <input
      checked={false}
      name="col"
      onChange={[Function]}
      type="radio"
      value="number"
    />
  </li>
  <li
    className="sort-column__item"
  >
    <input
      checked={false}
      name="col"
      onChange={[Function]}
      type="radio"
      value="distance"
    />
  </li>
</ul>
`;
